clients/apple/Shared/Stateful Logic/BillingService.swift

```diff
@@ -0,0 +1,203 @@
+import SwiftLockbookCore
+import SwiftUI
+import StoreKit
+
+public enum StoreError: Error {
+    case failedVerification
+    case noProduct
+}
+
+public enum PurchaseResult {
+    case success
+    case pending
+    case failure
+    case inFlow
+}
+
+public enum CancelSubscriptionResult {
+    case success
+    case appstoreActionRequired
+}
+
+
+let MONTHLY_SUBSCRIPTION_PRODUCT_ID = "basic.premium"
+
+class BillingService: ObservableObject {
+    let core: LockbookApi
+    
+    var pendingTransactionsListener: Task<Void, Error>? = nil
+
+    var maybeMonthlySubscription: Product? = nil
+    @Published var purchaseResult: PurchaseResult? = nil
+    @Published var cancelSubscriptionResult: CancelSubscriptionResult? = nil
+    
+    var showPurchaseToast: Bool = false
+    
+    init(_ core: LockbookApi) {
+        self.core = core
+        
+        if case .success(_) = core.getAccount() {
+            Task {
+                await launchBillingBackgroundTasks()
+            }
+        }
+    }
```

Comment:
What I'm thinking here is: I don't want the app to do this every time I open it. Ideally it only does this when I'm doing the billing flow. Costs some data, some battery, and maybe adds a hair of latency to startup (i know it's async). Can't we fire this off at some later point in the flow than every app startup?